x-pack/legacy/plugins/rollup/README.md

```diff
@@ -0,0 +1,51 @@
+
+Once the user has at least one rollup job configured, in the Index Pattern UI, the standard `Create index pattern` button will become a context menu with two options, one for regular index pattern and one for rollup index pattern. This is done by registering a [rollup config to index pattern creation extension points](public/index_pattern_creation/rollup_index_pattern_creation_config.js), in particular using the `getIndexPatternCreationOption()`. Once the user selects to create a rollup index pattern, this config also adjusts creation wizard in the following ways:
+1. Adds `Rollup` badge to list of matching indices that are rollup indices using `getIndexTags()`.
+2. Enforces index pattern rules using `checkIndicesForErrors()`. Rollup index patterns must match **one** rollup index, and optionally any number of regular indices. A rollup index pattern configured with one or more regular indices is affectionally known as a "hybrid" index pattern, and this allows the user to visualize historical (rollup) data and live (regular) data in the same visualization.
+3. Routes to this plugin's [rollup `_fields_for_wildcard` endpoint](server/routes/api/index_patterns.js), instead of the standard one, using `getFetchForWildcardOptions()`, so that the internal rollup data field names are mapped to the original field names.
+4. Writes additional information about what aggregations, fields, histogram interval, and date histogram interval and timezone to the rollup index pattern saved object using `getIndexPatternMappings()`. This collection of information referred to as its "capabilities".
+
+Once a rollup index pattern is created, it will be tagged with `Rollup` in the list of index patterns, and its details page will display capabilities information. This is done by registering [yet another config for index pattern list](public/index_pattern_list/rollup_index_pattern_list_config.js) extension points.
+
+## Create visualizations from rollup index patterns
+
+This plugin enables Kibana to create basic visualizations from rollup index patterns (note that this does not include TSVB, Vega, or Timelion). This is done by routing rollup index pattern queries to [Elasticsearch rollup search endpoint](https://www.elastic.co/guide/en/elasticsearch/reference/current/rollup-search.html). Querying this endpoint, as opposed to the standard index search, is needed because rollup indices use a different document structure than regular indices. In addition, visualization editor options are also limited to the aggregations and metrics that are available for the specified rollup index pattern.
```

Comment:
"This plugin enables the user to create visualizations from rollup data using the Visualize app, but not TSVB, Vega, or Timelion. When Visualize sends search requests, this plugin routes the requests to the [Elasticsearch rollup search endpoint](link), which is designed to search the special document structure within rollup indices. Users are limited to fewer visualization options based on the capabilities of the rollup index pattern they're visualizing."

Reply:
Also, are we sure TSVB doesn't handle rollups? I could have sworn I saw @alexwizp had a PR to support this. I can't seem to find it so maybe I'm imagining things, but it might be worth checking with him.

Reply:
"This plugin enables the user to create visualizations from rollup data using the Visualize app, excluding TSVB, Vega, and Timelion. When Visualize sends search requests, this plugin routes the requests to the Elasticsearch rollup search endpoint, which searches the special document structure within rollup indices. The visualization options available to users are based on the capabilities of the rollup index pattern they're visualizing."

Reply:
@cjcenizal there has been work in TSVB to support rollup indices, but that work is not handled by this plugin